src/routes/flush/flush.controller.spec.ts

```diff
@@ -0,0 +1,293 @@
+    });
+
+    it('should invalidate chains', async () => {
+      const chains = [chainBuilder().build(), chainBuilder().build()];
+      mockNetworkService.get.mockImplementation((url) => {
+        switch (url) {
+          case `${safeConfigApiUrl}/api/v1/chains`:
+            return Promise.resolve({
+              data: pageBuilder().with('results', chains).build(),
+            });
+          case `${safeConfigApiUrl}/api/v1/chains/${chains[0].chainId}`:
+            return Promise.resolve({ data: chains[0] });
+          case `${safeConfigApiUrl}/api/v1/chains/${chains[1].chainId}`:
+            return Promise.resolve({ data: chains[1] });
+          default:
+            return Promise.reject(`No matching rule for url: ${url}`);
+        }
+      });
+
+      // fill cache by requesting chains
+      await request(app.getHttpServer()).get('/v1/chains');
+      await request(app.getHttpServer()).get(`/v1/chains/${chains[0].chainId}`);
+      await request(app.getHttpServer()).get(`/v1/chains/${chains[1].chainId}`);
+
+      // check the cache is filled
+      expect(fakeCacheService.keyCount()).toBe(3);
+
+      // execute flush
+      await request(app.getHttpServer())
+        .post('/v2/flush')
+        .send(
+          invalidationPatternDtoBuilder().with('invalidate', 'chains').build(),
+        )
+        .expect(200);
+
+      // check the cache is empty
+      expect(fakeCacheService.keyCount()).toBe(0);
+    });
+
+    it('should invalidate contracts', async () => {
+      const chain = chainBuilder().build();
+      const contracts = [contractBuilder().build(), contractBuilder().build()];
+      mockNetworkService.get.mockImplementation((url) => {
+        switch (url) {
+          case `${safeConfigApiUrl}/api/v1/chains`:
+            return Promise.resolve({
+              data: pageBuilder().with('results', [chain]).build(),
+            });
+          case `${safeConfigApiUrl}/api/v1/chains/${chain.chainId}`:
+            return Promise.resolve({ data: chain });
+          case `${chain.transactionService}/api/v1/contracts/${contracts[0].address}`:
+            return Promise.resolve({ data: contracts[0] });
+          case `${chain.transactionService}/api/v1/contracts/${contracts[1].address}`:
+            return Promise.resolve({ data: contracts[1] });
+          default:
+            return Promise.reject(`No matching rule for url: ${url}`);
+        }
+      });
+
+      // fill cache by requesting contracts
+      await request(app.getHttpServer()).get(
+        `/v1/chains/${chain.chainId}/contracts/${contracts[0].address}`,
+      );
+      await request(app.getHttpServer()).get(
+        `/v1/chains/${chain.chainId}/contracts/${contracts[1].address}`,
+      );
+
+      // check the cache is filled
+      await Promise.all(
+        contracts.map(async (contract) =>
+          expect(
+            await fakeCacheService.get(
+              CacheRouter.getContractCacheDir(chain.chainId, contract.address),
```

Comment:
nitpick: I would use the expected string value here instead of going through the `CacheRouter`. Reason being that if we refactor `CacheRouter` in an unintended way this test would catch that.

Or maybe we should have some tests to specifically target this 🤔

Reply:
Yes I agree, I think having the expected string here is better 🙂

Reply:
Changed in [04c3cf8](https://github.com/5afe/safe-client-gateway-nest/pull/288/commits/04c3cf8e534137d209634a8f98ee83cd4e636576)